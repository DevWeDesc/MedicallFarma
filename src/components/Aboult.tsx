import React from "react";
import ImagesGrid from "../../public/ImagesGrid.png";
import Image from "next/image";

export const Aboult = () => {
  return (
    <div className="flex flex-col-reverse py-12 px-5 gap-5 lg:pt-0 lg:px-0 lg:grid lg:grid-cols-2 lg:gap-20 lg:py-0">
      <Image alt="" className="lg:min-h-[30rem] lg:w-full" src={ImagesGrid} />
      <div className="flex flex-col gap-3 lg:py-8 lg:pr-16 lg:gap-10">
        <div className="flex flex-col lg:gap-5">
          <h2 className="text-xl font-bold lg:text-5xl">
            Sobre a Medicall Farma
          </h2>
          <div className="text-xs font-light lg:text-lg">
            <p>
              Atuamos há mais de 10 anos no mercado de vendas e distribuição de
              materiais médicos e medicamentos hospitalares em todo o estado de
              São Paulo.
            </p>
            <p>
              Oferecemos a melhor experiência de atendimento aos nossos
              clientes, fornecedores e acionistas, dede o primeiro contato.
              Trabalhamos com um mix de Produtos.
            </p>
            <p>
              Diversificados das melhores marcas nacionais e internacionais
              garantimos total adilidade e pontualidade nas entregas através de
              um rigoroso processo de qualidade operacional.
            </p>
          </div>
        </div>
        <div className="flex flex-col lg:gap-5">
          <h2 className="text-xl font-bold lg:text-5xl">Nossa Missão</h2>
          <p className="text-xs font-light lg:text-lg">
            A Medicall Farma tem como missão contribuir com a melhoria da vida e
            saúde das pessoas, por meio de fornecimento de materiais médicos e
            medicamentos hospitalares de alta qualidade às instituições de saúde
            privadas e públicas do estado de São Paulo.
          </p>
        </div>
      </div>
    </div>
  );
};
