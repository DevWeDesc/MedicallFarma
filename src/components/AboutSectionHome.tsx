import React from "react";
import imageMedicall from "../../public/imagemedicall.jpeg";
import Image from "next/image";
import Link from "next/link";
import { AnimationScroll } from "@/animations/AnimationScroll";

export const AboutSectionHome = () => {
  return (
    <AnimationScroll>
      <div className="flex flex-col justify-center relative items-center bg-black text-white">
        <Image
          alt=""
          className="w-full object-cover h-80 opacity-60"
          src={imageMedicall}
        />
        <div className="absolute h-full w-full flex flex-col justify-center gap-8 items-center">
          <p className="text-2xl font-bold px-5 text-center">
            Conhe√ßa mais sobre a MEDICALL FARMA
          </p>
          <Link
            className="text-center border w-auto px-8 py-2 rounded-lg hover:font-semibold hover:border-[2px] transition-all"
            href="about"
          >
            Saiba Mais!
          </Link>
        </div>
      </div>
    </AnimationScroll>
  );
};
