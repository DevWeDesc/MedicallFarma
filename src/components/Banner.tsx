"use client";
import React from "react";
// import ImageBanner from "../../public/ImageBanner.png";
import ImageBanner from "../../public/GroupProductsBanner.svg";
import Image from "next/image";
import { motion } from "framer-motion";

export const Banner = () => {
  return (
    <div
      // className="bg-grayDark px-5 flex flex-col-reverse gap-8 py-16 xl:grid xl:grid-cols-2 xl:items-center xl:gap-28 xl:pl-28 xl:pt-24 2xl:pl-80"
      className="bg-grayDark px-5 flex flex-col gap-8 py-16 sm:pt-24 xl:items-center xl:gap-12 xl:pt-24"
    >
      <div className="text-white flex flex-col gap-8">
        <h2 className="max-w-[387px] m-auto text-2xl sm:text-3xl xl:max-w-4xl font-light text-center xl:text-5xl">
          Encontre os melhores Produtos Hospitalares com os Melhores Pre√ßos aqui
          na <span className="font-bold">Medicall Farma!</span>
        </h2>
        {/* <Link
          href="#productRecomended"
          className="xl:max-w-md text-center rounded-full py-3 px-10 font-bold text-xl hover:bg-greenDark transition-all bg-greenDefault"
        >
          Procurar Medicamentos
        </Link> */}
      </div>
      <motion.div
        initial={{ scale: 0 }}
        animate={{ scale: 1 }}
        whileHover={{ scale: 1.1 }}
        transition={{ duration: 1.5 }}
      >
        <Image alt="" className="m-auto" src={ImageBanner} />
      </motion.div>
      {/* <Image alt="" src={ImageBanner} /> */}
    </div>
  );
};
