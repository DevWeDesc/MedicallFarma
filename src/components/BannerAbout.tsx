import Image from "next/image";
import React from "react";
import ImageBannerAbout from "../../public/Rectangle.png";
import ImageBannerAboutMobile from "../../public/ImageAboutMobile.png";

export const BannerAbout = () => {
  return (
    <div className="relative flex flex-col justify-center items-center pt-22">
      <Image
        alt=""
        className="hidden lg:block w-full h-auto"
        src={ImageBannerAbout}
      />
      <Image alt="" src={ImageBannerAboutMobile} className="lg:hidden w-full" />
      <div className="absolute text-white flex flex-col items-center w-full gap-8 px-4">
        <h2 className="max-w-[387px] text-center text-2xl font-bold lg:max-w-4xl lg:m-auto xl:text-5xl">
          Missão de Saúde: Fornecendo Excelência em Medicamentos e Materiais
          Médicos
        </h2>
        <p className="text-sm lg:text-base lg:max-w-4xl lg:m-auto text-center font-light">
          Bem-vindo à MEDICALL FARMA, sua parceira dedicada na entrega de
          soluções de saúde excepcionais. Com uma visão centrada no bem-estar,
          estamos comprometidos em fornecer medicamentos e materiais médicos de
          qualidade superior, promovendo a excelência no cuidado médico.
          Descubra mais sobre nossa história, valores e o compromisso que temos
          em impulsionar a saúde e o progresso na comunidade médica.
        </p>
      </div>
    </div>
  );
};
