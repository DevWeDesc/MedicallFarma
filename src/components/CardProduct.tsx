"use client";
import React, { useEffect, useState } from "react";
import { IProductsData } from "../../types/types";
import Image from "next/image";

export const CardProduct = ({
  image,
  name,
  content,
  supplier,
  categories,
}: IProductsData) => {
  const [configCard, setConfigCard] = useState("");
  const [contentProduct, setContentProduct] = useState("");

  useEffect(() => {
    if (location) {
      location.pathname != "/" && setConfigCard("large");
    }
    configCard === "large"
      ? setContentProduct(content)
      : setContentProduct(content?.substring(0, 83));
  }, [configCard]);

  return (
    <div
      className={`flex flex-col text-left ${
        configCard === "large" ? "gap-4 items-center" : "gap-2"
      } gap-2`}
    >
      <Image alt="" src={image} />
      <div className={`${configCard === "large" && "flex flex-col gap-2"}`}>
        <p
          className={`font-medium min-h-[32px] ${
            configCard === "large" ? "text-2xl" : "text-xs "
          }`}
        >
          {name}
        </p>
        <p
          className={`font-light pr-3 ${
            configCard === "large" ? "text-sm opacity-70" : "text-xs"
          }`}
        >
          {contentProduct}
        </p>
        <p
          className={`font-medium  ${
            configCard === "large" ? "text-base" : "text-xs"
          }`}
        >
          {supplier}
        </p>
        {configCard === "large" && (
          <div>
            <button>Solicitar um Or√ßamento</button>
            <p>
              Categorias:{" "}
              {categories?.map((item, index) => (
                <span key={index}>{item}</span>
              ))}
            </p>
          </div>
        )}
      </div>
    </div>
  );
};
