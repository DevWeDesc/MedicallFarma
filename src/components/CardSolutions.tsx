"use client";
import React, { useEffect, useState } from "react";
import { ICardSoluctions } from "../../types/types";
import Image from "next/image";
import image from "next/image";

export const CardSolutions = ({ image, title }: ICardSoluctions) => {
  const [bgColorCard, setBgColorCard] = useState("");
  useEffect(() => {
    if (title === "Medicamentos") {
      setBgColorCard("bg-blueDark");
    } else if (title === "Dietas Enterais") {
      setBgColorCard("bg-yellowDefault");
    } else {
      setBgColorCard("bg-greenDark");
    }
  }, []);

  return (
    <div
      className={`relative px-2 pt-3 pb-12 rounded-lg ${bgColorCard} hover:bg-opacity-90 transition-all`}
    >
      <p className="text-sm font-semibold">{title}</p>
      <button className="text-[10px] bg-transparent border-b pb-[2px] border-white">
        Conhe√ßa agora
      </button>
      <Image alt="" className="absolute right-2 bottom-2" src={image} />
    </div>
  );
};
