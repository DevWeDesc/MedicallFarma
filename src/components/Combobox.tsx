"use client";
import Image from "next/image";
import Link from "next/link";
import React, { useState } from "react";
import { CategoriesData } from "../../data/data";
import { ICategories } from "../../types/types";
import { CiPlay1 } from "react-icons/ci";

export const Combobox = () => {
  const [comboOpen, setComboOpen] = useState(false);
  const [categorySelected, setCategorySelected] = useState<ICategories>();

  const handleComboOpen = (item: ICategories) => {
    if (comboOpen) {
      setComboOpen(false);
      setCategorySelected(item);
    } else {
      setCategorySelected(item);
      setComboOpen(true);
    }
  };

  return (
    <div className="hidden xl:flex flex-col pt-20 ">
      <div className="bg-grayDark text-white flex gap-10 justify-center">
        {CategoriesData.map((item, index) => (
          <button
            key={index}
            onMouseLeave={() => handleComboOpen(item)}
            onMouseOver={() => handleComboOpen(item)}
            className={`p-8 bg-grayDark hover:font-bold transition-all ${
              categorySelected?.name === item.name && "font-bold"
            }`}
          >
            {item.name}
          </button>
        ))}
      </div>
      <div
        className="relative flex w-full justify-center"
        onMouseLeave={() => {
          setComboOpen(false);
          setCategorySelected(categorySelected);
        }}
        onMouseOver={() => {
          setComboOpen(true);
          setCategorySelected(categorySelected);
        }}
      >
        <div
          className={`${
            comboOpen && categorySelected
              ? "absolute bg-grayLight py-10 px-6 rounded-md grid grid-cols-2 text-white transition-all"
              : "absolute h-0 transition-all overflow-hidden"
          }`}
        >
          <div className="flex flex-col gap-6 ">
            {categorySelected?.subCategories.map((item, index) => (
              <Link
                href={`${categorySelected.name}/${item.title}`}
                key={index}
                className="flex items-center gap-3 opacity-90 hover:opacity-100 hover:font-semibold transition-all"
              >
                <CiPlay1 />
                <p>{item.title}</p>
              </Link>
            ))}
          </div>
          {categorySelected && (
            <Image
              alt=""
              className="max-w-sm"
              src={categorySelected?.image}
            ></Image>
          )}
        </div>
      </div>
    </div>
  );
};
