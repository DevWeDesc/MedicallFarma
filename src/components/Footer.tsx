import Image from "next/image";
import React from "react";
import Logo from "../../public/Logo.png";
import { NavData } from "../../data/data";

export const Footer = () => {
  return (
    <div className="bg-grayDark py-16 flex flex-col items-center gap-16 text-white">
      <Image alt="" src={Logo} />
      <div className="flex flex-col gap-4">
        <h3 className="text-xl font-bold">Mapa do Site</h3>
        <div className="flex flex-col gap-2">
          {NavData.map((item, index) => (
            <p key={index} className="text-sm font-light">
              {item.name}
            </p>
          ))}
        </div>
      </div>
      <div className="flex flex-col gap-4">
        <h3 className="text-xl font-bold text-center">Nossas Redes Sociais</h3>
        <div className="flex gap-2">
          {NavData.map((item, index) => (
            <p key={index} className="text-sm font-light">
              {item.name}
            </p>
          ))}
        </div>
      </div>
      <p className="text-center text-sm">
        Â© Copyright 2023 | Desenvolvido por WeDesc | Todos os Direitos
        Reservados
      </p>
    </div>
  );
};
