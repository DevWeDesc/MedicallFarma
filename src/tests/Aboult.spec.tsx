/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Aboult } from "@/components/Aboult";
import "intersection-observer";

describe("Aboult render", () => {
  it("renders Aboult unchanged", () => {
    const { container } = render(<Aboult />);
    expect(container).toMatchSnapshot();
  });

  it("renders Aboult text", () => {
    const { getByText, getByRole } = render(<Aboult />);

    expect(getByText("Sobre a Medicall Farma")).toBeInTheDocument();
    expect(
      getByText(
        "Atuamos há mais de 10 anos no mercado de vendas e distribuição de materiais médicos e medicamentos hospitalares em todo o estado de São Paulo."
      )
    ).toBeInTheDocument();
    expect(
      getByText(
        "Oferecemos a melhor experiência de atendimento aos nossos clientes, fornecedores e acionistas, dede o primeiro contato. Trabalhamos com um mix de Produtos."
      )
    ).toBeInTheDocument();
    expect(
      getByText(
        "Diversificados das melhores marcas nacionais e internacionais garantimos total adilidade e pontualidade nas entregas através de um rigoroso processo de qualidade operacional."
      )
    ).toBeInTheDocument();
  });

  it("renders Aboult text", () => {
    const { getByRole } = render(<Aboult />);

    expect(getByRole("img")).toBeInTheDocument();
  });
});
