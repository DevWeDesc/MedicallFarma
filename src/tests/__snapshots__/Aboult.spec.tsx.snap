// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aboult render renders Aboult unchanged 1`] = `
<div>
  <div
    class="flex flex-col-reverse py-12 px-5 gap-5 lg:pt-0 lg:px-0 lg:grid lg:grid-cols-2 lg:gap-20 lg:py-0"
  >
    <img
      alt=""
      class="w-full lg:min-h-[30rem] lg:w-full lg:h-auto"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
    <div
      class="flex flex-col gap-4 lg:py-8 lg:pr-16 lg:gap-10 2xl:pr-80 xl:justify-center"
    >
      <div
        class="flex flex-col lg:gap-5"
      >
        <h2
          class="text-4xl font-bold lg:text-5xl"
        >
          Sobre a Medicall 
          <br />
           Farma
        </h2>
        <div
          class="text-xs font-light lg:text-lg flex flex-col gap-1 lg:gap-3"
        >
          <p>
            Atuamos há mais de 10 anos no mercado de vendas e distribuição de materiais médicos e medicamentos hospitalares em todo o estado de São Paulo.
          </p>
          <p>
            Oferecemos a melhor experiência de atendimento aos nossos clientes, fornecedores e acionistas, dede o primeiro contato. Trabalhamos com um mix de Produtos.
          </p>
          <p>
            Diversificados das melhores marcas nacionais e internacionais garantimos total adilidade e pontualidade nas entregas através de um rigoroso processo de qualidade operacional.
          </p>
          <p>
            Nossos colaboradores não medem esforços para alcançar a plena satisfação de nossos clientes e parceiros. Agilidade, preticidade, inovação e ética são compromissos assumidos pela MEDICALL FARMA.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
